misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/jdi/StackFrameProxyImpl.java

```diff
@@ -128,17 +129,31 @@ protected void clearCaches() {
     @Override
     public StackFrame getStackFrame() throws JdiProxyException {
         checkValid();
-        if (myStackFrame == null) {
+        if (myStackFrame != null) {
+            return myStackFrame;
+        }
+        IncompatibleThreadStateException error = null;
+        for (int attempt = 0; attempt < RETRY_COUNT; attempt++) {
             try {
                 final ThreadReference threadRef = myThreadProxy.getThreadReference();
                 myStackFrame = threadRef.frame(getFrameIndex());
-            } catch (IndexOutOfBoundsException | IncompatibleThreadStateException e) {
+                return myStackFrame;
+            } catch (IncompatibleThreadStateException e) {
+                error = e;
+                clearCaches();
+                try {
+                    Thread.sleep(10);
+                } catch (InterruptedException ignored) {
```

Comment:
Are we intentionally ignoring the catch clause here?

Reply:
Yes we are and thats why I've marked the variable name as `ignored`. Because If the sleep is interrupted, it's not critical to the overall operation and the code can continue with the next retry attempt regardless.